arcgis-ios-sdk-samples/Scenes/Animate image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,228 @@
+    let imageOverlay = AGSImageOverlay()
+    /// The frame rate in the unit of frame per second.
+    var frameRate = 60.0
+    /// An array to hold the URLs for the overlay images.
+    var imageURLs: [URL] = []
+    /// An envelope of the pacific southwest sector for displaying the image frame.
+    var pacificSouthwestEnvelope: AGSEnvelope!
+    /// A timer which repeatedly calls setImageFrame for a given period.
+    var frameRateTimer: Timer?
+    /// A boolean which indicates whether the animation is playing or paused.
+    var isAnimating = false {
+        didSet {
+            playPauseButton.title = isAnimating ? "Pause" : "Play"
+        }
+    }
+    
+    // MARK: Initialize scene and animation
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        // Create a tiled layer from World Dark Gray Base REST service.
+        let basemapTileURL = URL(string: "https://services.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer")!
+        let worldDarkGrayBasemap = AGSArcGISTiledLayer(url: basemapTileURL)
+        // Create an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        let scene = AGSScene(basemap: AGSBasemap(baseLayer: worldDarkGrayBasemap))
+        // Create a surface and add it to the scene.
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        scene.baseSurface = surface
+        // Create an envelope of the pacific southwest sector for displaying the image frame.
+        let pointForImageFrame = AGSPoint(
+            x: -120.0724273439448,
+            y: 35.131016955536694,
+            spatialReference: .wgs84()
+        )
+        pacificSouthwestEnvelope = AGSEnvelope(
+            center: pointForImageFrame,
+            width: 15.09589635986124,
+            height: -14.3770441522488
+        )
+        return scene
+    }
+    
+    func startAnimation(_ fps: Double) {
+        // Invalidate timer to stop previous ongoing animation.
+        frameRateTimer?.invalidate()
+        // Duration or interval between frames in seconds.
+        let duration = 1 / Double(fps)
+        let timer = Timer(timeInterval: duration, repeats: true) { [weak self] _ in
+            self?.setImageFrame()
+        }
+        self.frameRateTimer = timer
+        // Add the timer to common mode run loop, so the timer is not effected by UI events.
+        RunLoop.main.add(timer, forMode: .common)
+    }
+    
+    /// Load the URLs for the overlay images.
+    ///
+    /// - Parameter completion: If it loads successfully, enable the UI components in the completion closure.
+    func loadImageURLs(completion: () -> Void) {
```

Comment:
Initially I write this non-escaping completion closure for updating UI after loading complete, so that I don't incorporate unrelated logic into loading data function. I'll try with another way to put it.